### PR TITLE
Implement ReadProfileCommand, also add filter profile by ID feature to ListProfilesCommand

### LINKED ISSUE #211: Get a Provisioning Profile by UUID
# 🗣 Context

When fetching specific profiles.

# 💬 Narrative

As a user
I want to be able to fetch a provisioning profile by uuid
So that I can download a single profile

# 📝 Notes

If I know the UUID of a provisioning profile I think it would be useful to be able to download just that single profile.

# 🎨 Design

```
asc profiles read [--api-issuer <uuid>] [--api-key-id <keyid>] [--csv] [--json] [--table] [--yaml] [--download-path <path>] <uuid>
```

# ✅ Acceptance Criteria

**GIVEN** there is a provisioning profile with the specified uuid
**WHEN** I request that profile
**THEN** The profile information is displayed

**GIVEN** there is a provisioning profile with the specified uuid
**WHEN** I request that profile and I specify a valid `--download-path`
**THEN** The profile is saved to the specified `<path>`

### PR DESCRIPTION
**Pull request by DechengMa**
Closing #211 

# 📝 Summary of Changes

Changes proposed in this pull request:

- Add `Id` as attributes to `Profile` 
- Closes #211 by Implement `ReadProfileCommand`
- Add `--filter` option to `ListProfilesCommand`

# ⚠️ Items of Note

This PR will expose the internal Id to user

# 🧐🗒 Reviewer Notes

## 💁 Example
ID will be shown in this command:
```
asc profiles list
+------------+--------------------------------------+---------------------+----------+--------+---------------------+----------------------+----------------------+
| ID         | UUID                                 | Name                | Platform | State  | Type                | Created Date         | Expiration Date      |
+------------+--------------------------------------+---------------------+----------+--------+---------------------+----------------------+----------------------+
```

The id can be accepted as filter in `ListProfilesCommand`:

``` 
swift run asc profiles list --filter-id <filter-id>
```

Also implemented `ReadProfileCommand`
```
asc profiles read <id> [--download-path <path>]
```

## 🔨 How To Test

```
swift run asc profiles list 1N2Z3LUSP8 --download-path ./../
```
```
swift run asc profiles read 1N2Z3LUSP8 --download-path ../../
```

### CODE REVIEW

File: Sources/AppStoreConnectCLI/Model/Profile.swift [hunk 1]
```diff
@@ -7,15 +7,12 @@ import SwiftyTextTable
...
-        self.init(apiProfile.attributes!)
-    }
-
-    init(_ attributes: AppStoreConnect_Swift_SDK.Profile.Attributes) {
+        let attributes = apiProfile.attributes!
         self.init(
             name: attributes.name,
             platform: attributes.platform?.rawValue,
             profileContent: attributes.profileContent,
-            uuid: attributes.uuid,
```
Comment (orj) [inline Sources/AppStoreConnectCLI/Model/Profile.swift:18]: The UUID is important too in profiles.  As it is usually the filename used to save the profile.

Comment (DechengMa) [reply Sources/AppStoreConnectCLI/Model/Profile.swift:18]: As UUID can't be used to get profile, how about showing ID and UUID together?
```
asc profiles list
+------------+------------------------------+----------+--------+---------------+----------------------+----------------------+
| ID         |  UUID         | Name                         | Platform | State  | Type          | Created Date         | Expiration Date      |
+------------+------------------------------+----------+--------+---------------+----------------------+----------------------+
```

Comment (orj) [reply Sources/AppStoreConnectCLI/Model/Profile.swift:18]: Showing them together is fine.

Comment (DechengMa) [reply Sources/AppStoreConnectCLI/Model/Profile.swift]: ✅

**Comment by orj** [comment]
The issue in question here is actually asking for a seperate command @DechengMa.  For consistency we should have a `read` command as well as a list command.  And also for consistency if we want to filter items in the list they should be provided via `--filter` arguments.

**Comment by orj** [comment]
@DechengMa This doesn't actually quite do what #211 was asking.  That issue was asking for a filter on UUID.  Which technically can't be done with the API.  But we could do it on the client side.  But having looked back at that issue I'm not sure why I was even asking for that feature. I didn't record my motivations well enough.

**Review by orj — APPROVED** [review]
I think we'll merge this.

**Pull request merged**